File: src/CSPAF/CSPAgents/CSPAgentDetails/CSPVerifyAgentCommunication.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace actions.CSP.CSPAgents.CSPAgentDetails
{
    class CSPVerifyAgentCommunication
    {
        public void run(Dictionary<string, object> Params)
        {
            var navigationController = CSPBrowser.Application.WindowController.NavigationController;
            var controller = (CSPAgentDetailsController)navigationController.VisibleViewController;
            controller.VerifyAgentCommunication();
        }
    }
}

File: src/CSPAF/CSPAgents/CSPAgentDetails/CSPVerifyAgentCommunication.java
package CSPAF.CSPAgents.CSPAgentDetails;

import java.io.*;
import java.util.*;

import com.infor.cloudsuite.portal.core.services.*;
import com.infor.cloudsuite.portal.core.controllers.*;
import com.infor.cloudsuite.portal.singletenant.agents.controllers.*;

import org.openqa.selenium.NotFoundException;

class CSPVerifyAgentCommunication
{
    public void run(HashMap<String, Object> params) throws IOException, NotFoundException, InterruptedException
    {
        CSPNavigationController navigationController = CSPBrowser.getApplication().getWindowController().getNavigationController();
        CSPAgentDetailsController controller = (CSPAgentDetailsController)navigationController.getVisibleViewController();
        controller.verifyAgentCommunication();
    }
}